framework/src/dirackernels/VectorPostprocessorPointSource.C

```diff
@@ -0,0 +1,80 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "VectorPostprocessorPointSource.h"
+#include "MooseUtils.h"
+#include "DelimitedFileReader.h"
+
+registerMooseObject("MooseApp", VectorPostprocessorPointSource);
+
+defineLegacyParams(VectorPostprocessorPointSource);
+
+InputParameters
+VectorPostprocessorPointSource::validParams()
+{
+  InputParameters params = DiracKernel::validParams();
+
+  params.addClassDescription("Apply a point load defined by VectorPostprocessor.");
+
+  params.addParam<VectorPostprocessorName>(
+      "vector_postprocessor",
+      "The name of the VectorPostprocessor containing positions and corresponding");
+  params.addParam<std::string>("x_coord_name", "x", "name of column containing x coordinates.");
+  params.addParam<std::string>("y_coord_name", "y", "name of column containing y coordinates.");
+  params.addParam<std::string>("z_coord_name", "z", "name of column containing z coordinates.");
+  params.addParam<std::string>("value_name", "value", "name of column containing values.");
+
+  params.addParam<bool>(
+      "use_broadcast",
+      false,
+      "Causes this diracKernel to use a broadcasted version of the vector instead "
+      "of a scattered version of the vector (the default).  This is slower - but "
+      "is useful for debugging and testing");
+
+  return params;
+}
+
+VectorPostprocessorPointSource::VectorPostprocessorPointSource(const InputParameters & parameters)
+  : DiracKernel(parameters),
+    VectorPostprocessorInterface(this),
+    _use_broadcast(getParam<bool>("use_broadcast")),
+    _vpp_values(getVectorPostprocessorValue(
+        "vector_postprocessor", getParam<std::string>("value_name"), _use_broadcast))
+{
+}
+
+void
+VectorPostprocessorPointSource::addPoints()
+{
+  const auto xcoord(getVectorPostprocessorValue(
```

Comment:
this could be gotten as a member in the initializer list, or is there a reason fro re-getting this each time step?

Reply:
If that was the case we couldn't have moving point sources

Reply:
I made the names member data and they are added to the initializer list.